### PR TITLE
Take into consideration child organizations for dashboard admins

### PR DESCRIPTION
**Pull request by marcospri**
This is the last part needed for: 
- https://github.com/hypothesis/lms/issues/6480 

and also needed for: 

- https://github.com/hypothesis/lms/issues/6378

Grant not only access to one organization but all its children

### Testing
- Create a new organization http://localhost:8001/admin/org/new

- Assign your local "University of hypothesis" to that new org (make it a child)

- (In `main`) Access the staff dashboard from the new organization, you'll get an empty dashboard as there no activity on that organization.

- Switch to `admin-child-organizations` you'll get now all the activity from the children org.

### CODE REVIEW

File: lms/services/dashboard.py [hunk 1]
```diff
@@ -9,7 +9,11 @@
 
 class DashboardService:
     def __init__(
-        self, request, assignment_service, course_service, organization_service
+        self,
+        request,
+        assignment_service,
+        course_service,
+        organization_service: OrganizationService,
```
Comment (marcospri) [inline lms/services/dashboard.py:16]: Adding a type anno here.

File: lms/services/dashboard.py [hunk 2]
```diff
@@ -69,11 +73,19 @@ def get_request_course(self, request):
 
     def get_organizations_by_admin_email(self, email: str) -> list[Organization]:
         """Get a list of organizations where the user with email `email` is an admin in."""
-        return self._db.scalars(
-            select(Organization)
-            .join(DashboardAdmin)
+        organization_ids = []
+
+        for org_id in self._db.scalars(
```
Comment (marcospri) [inline lms/services/dashboard.py:78]: For each of the organization where this user is a top level admin.

File: lms/services/dashboard.py [hunk 3]
```diff
@@ -69,11 +73,19 @@ def get_request_course(self, request):
...
+            select(DashboardAdmin.organization_id)
             .where(DashboardAdmin.email == email)
             .distinct()
+        ).all():
+            organization_ids.extend(
+                self._organization_service.get_hierarchy_ids(org_id)
```
Comment (marcospri) [inline lms/services/dashboard.py:84]: Fetch all the ids of the hierarchy

File: lms/services/dashboard.py [hunk 4]
```diff
@@ -104,7 +116,13 @@ def get_request_admin_organizations(self, request) -> list[Organization]:
             if not organization:
                 raise HTTPNotFound()
 
-            return [organization]
+            return self._db.scalars(
+                select(Organization).where(
+                    Organization.id.in_(
+                        self._organization_service.get_hierarchy_ids(organization.id)
```
Comment (marcospri) [inline lms/services/dashboard.py:122]: Same thing for staff members

**Review by acelaya — APPROVED** [review]

**Pull request merged**